main/src/cgeo/geocaching/ui/CalculatorVariable.java

```diff
@@ -0,0 +1,304 @@
+        if (isCacheDirty())
+            EvaluateDouble(dependantVariables);
+
+        return getCachedString();
+    }
+
+    private double EvaluateDouble(final List<CalculatorVariable> dependantVariables) {
+        if (isCacheDirty()) {
+            String expression = getExpression();
+
+            if (dependantVariables != null) {
+                for (final CalculatorVariable depVar : dependantVariables)
+                    expression = expression.replace("" + depVar.getName(), "(" + depVar.EvaluateDouble(null) + ")");
+            }
+
+            try {
+                setCachedValue(eval(expression));
+            } catch (final Exception e) {
+                setCachedValue(Double.NaN);
+            }
+        }
+
+        return getCachedValue();
+    }
+
+    // This simple evaluation algorithm was created by user 'Boann' and released to the public domain on Stack Overflow:
+    // https://stackoverflow.com/questions/3422673/evaluating-a-math-expression-given-in-string-form
+    // I added the '%' operator myself.
+    // Note that the 'sqrt', 'cos', 'sin' and 'tan' functions don't work in this implementation as
+    // expected as the individual letters are automatically replaced with their corresponding
+    // expressions as they are interpreted as being individual variables.
+    private static double eval(final String str) {
+        return new Object() {
+            int pos = -1, ch;
+
+            void nextChar() {
+                ch = (++pos < str.length()) ? str.charAt(pos) : -1;
+            }
+
+            boolean eat(final int charToEat) {
+                while (ch == ' ') nextChar();
+                if (ch == charToEat) {
+                    nextChar();
+                    return true;
+                }
+                return false;
+            }
+
+            double parse() {
+                nextChar();
+                final double x = parseExpression();
+                if (pos < str.length()) throw new RuntimeException("Unexpected: " + (char)ch);
+                return x;
+            }
+
+            // Grammar:
+            // expression = term | expression `+` term | expression `-` term
+            // term = factor | term `*` factor | term `/` factor
+            // factor = `+` factor | `-` factor | `(` expression `)`
+            //        | number | functionName factor | factor `^` factor
+
+            double parseExpression() {
+                double x = parseTerm();
+                for (;;) {
+                    if      (eat('+')) x += parseTerm(); // addition
+                    else if (eat('-')) x -= parseTerm(); // subtraction
+                    else return x;
+                }
+            }
+
+            double parseTerm() {
+                double x = parseFactor();
+                for (;;) {
+                    if      (eat('*')) x *= parseFactor(); // multiplication
+                    else if (eat('/')) x /= parseFactor(); // division
```

Comment:
I was hoping to leave this method as is.

It's fairly neat and compact and is actually someone else's code.